src/CodeGen_Internal.cpp

```diff
@@ -123,14 +123,18 @@ StructType *Closure::build_type(LLVMContext *context) {
     return struct_t;
 }
 
-void Closure::pack_struct(Value *dst, const Scope<Value *> &src, IRBuilder<> *builder) {
-    // dst should be a pointer to a struct of the type returned by build_type
+void Closure::pack_struct(llvm::Type *type, Value *dst, const Scope<Value *> &src, IRBuilder<> *builder) {
+    // type, type of dst should be a pointer to a struct of the type returned by build_type
     int idx = 0;
     LLVMContext &context = builder->getContext();
     vector<string> nm = names();
     vector<llvm::Type*> ty = llvm_types(&context);
     for (size_t i = 0; i < nm.size(); i++) {
+#if LLVM_VERSION >= 37
+        Value *ptr = builder->CreateConstInBoundsGEP2_32(type, dst, 0, idx);
+#else
         Value *ptr = builder->CreateConstInBoundsGEP2_32(dst, 0, idx);
```

Comment:
type is going to be unused here if LLVM_VERSION is less than 37. This causes a warning. Not sure of the best way to fix that. Could conditionally comment out the name of the argument in the prototype, or try to do something to force the type to be used in the other branch of the #ifdef. (It is also possible to write templates that match the decl that exists in some circumstances, which might be productive, but I have not investigated.)